rclcpp/src/rclcpp/signal_handler.cpp

```diff
@@ -151,98 +214,57 @@ SignalHandler::~SignalHandler()
   } catch (const std::exception & exc) {
     RCLCPP_ERROR(
       get_logger(),
-      "caught %s exception when uninstalling the sigint handler in rclcpp::~SignalHandler: %s",
+      "caught %s exception when uninstalling signal handlers in rclcpp::~SignalHandler: %s",
       rmw::impl::cpp::demangle(exc).c_str(), exc.what());
   } catch (...) {
     RCLCPP_ERROR(
       get_logger(),
-      "caught unknown exception when uninstalling the sigint handler in rclcpp::~SignalHandler");
+      "caught unknown exception when uninstalling signal handlers in rclcpp::~SignalHandler");
   }
 }
 
 SignalHandler::signal_handler_type
-SignalHandler::set_signal_handler(
-  int signal_value,
-  const SignalHandler::signal_handler_type & signal_handler)
+SignalHandler::get_old_signal_handler(int signum)
 {
-  bool signal_handler_install_failed;
-  SignalHandler::signal_handler_type old_signal_handler;
+  if (SIGINT == signum) {
+    return old_sigint_handler_;
+  } else if (SIGTERM == signum) {
+    return old_sigterm_handler_;
+  }
 #if defined(RCLCPP_HAS_SIGACTION)
-  ssize_t ret = sigaction(signal_value, &signal_handler, &old_signal_handler);
-  signal_handler_install_failed = (ret == -1);
+  SignalHandler::signal_handler_type ret;
+  memset(&ret, 0, sizeof(ret));
```

Comment:
We probably shouldn't use a bare `memset` here, instead maybe the C11 variant or the Windows specific "safe" version:

https://en.cppreference.com/w/c/string/byte/memset

Ideally, we'd have our own util function to do the right thing in the right places, like we do for other C calls, like `snprintf` in `rcutils`.

The reason I mention this is that I'm currently fixing clang-tidy warnings and this is one of them (using normal `memset` rather than `memset_s`).

Reply:
I know we were using it before, I just wanted to point it out. I'll have to change that after this is merged, which is fine.

Reply:
The c11 variant (memset_s) is not supported by linux (all the bound checked library functions are "optional" in the c11 standard).
Could you show me a code snippet of the approach you're using so I apply the same one here?